boards/entrypoint.sh

```diff
@@ -0,0 +1,115 @@
+}
+
+AZURE_BOARDS_TYPE="${AZURE_BOARDS_TYPE:-Feature}"
+AZURE_BOARDS_CLOSED_STATE="${AZURE_BOARDS_CLOSED_STATE:-Closed}"
+AZURE_BOARDS_REOPENED_STATE="${AZURE_BOARDS_REOPENED_STATE:-New}"
+AZURE_DEVOPS_URL="https://dev.azure.com/${AZURE_BOARDS_ORGANIZATION}/"
+
+vsts configure --defaults instance="${AZURE_DEVOPS_URL}" project="${AZURE_BOARDS_PROJECT}"
+
+vsts login --token "${AZURE_BOARDS_TOKEN}"
+
+GITHUB_EVENT=$(jq --raw-output 'if .comment != null then "comment" else empty end' "$GITHUB_EVENT_PATH")
+GITHUB_EVENT=${GITHUB_EVENT:-$(jq --raw-output 'if .issue != null then "issue" else empty end' "$GITHUB_EVENT_PATH")}
+GITHUB_ACTION=$(jq --raw-output .action "$GITHUB_EVENT_PATH")
+GITHUB_ISSUE_NUMBER=$(jq --raw-output .issue.number "$GITHUB_EVENT_PATH")
+GITHUB_ISSUE_HTML_URL=$(jq --raw-output .issue.html_url "$GITHUB_EVENT_PATH")
+AZURE_BOARDS_TITLE=$(jq --raw-output .issue.title "$GITHUB_EVENT_PATH")
+AZURE_BOARDS_DESCRIPTION=$(jq --raw-output .issue.body "$GITHUB_EVENT_PATH" | parse_markdown)
+
+TRIGGER="${GITHUB_EVENT}/${GITHUB_ACTION}"
+
+case "$TRIGGER" in
+"issue/opened")
+    # If there's a GitHub issue label configured then don't create a
+    # corresponding Azure Boards work item.  Wait for a labelled event
+    # to create the work item.
+    if [ -z "$ISSUE_LABEL" ]; then
+        create_work_item
+    fi
+    ;;
+
+"issue/labeled")
+    # If there's a GitHub issue label configured then see if that was the
+    # label applied.  If so, create a new Azure Boards work item to
+    # correspond to this issue.
+    NEW_LABEL=$(jq --raw-output .label.name "$GITHUB_EVENT_PATH")
+
+    if [ ! -z "$ISSUE_LABEL" ] && [[ "$NEW_LABEL" == "$ISSUE_LABEL" ]]; then
+        create_work_item
+    fi
+    ;;
+
+"issue/reopened"|"issue/closed")
+    [[ "$GITHUB_ACTION" = "reopened" ]] && \
+        NEW_STATE="$AZURE_BOARDS_REOPENED_STATE" || \
+        NEW_STATE="$AZURE_BOARDS_CLOSED_STATE"
+
+    echo "Looking for work items with tag 'Issue ${GITHUB_ISSUE_NUMBER}'..."
+
+    for ID in $(work_items_for_issue); do
+        echo "Setting work item ${ID} to state ${NEW_STATE}..."
+        RESULTS=$(vsts work item update --id "$ID" --state "$NEW_STATE")
+
+        RESULT_STATE=$(echo "${RESULTS}" | jq --raw-output '.fields["System.State"]')
+        echo "Work item ${ID} is now ${RESULT_STATE}"
+    done
+    ;;
+
+"comment/created")
+    echo "Looking for work items with tag 'Issue ${GITHUB_ISSUE_NUMBER}'..."
+
+    for ID in $(work_items_for_issue); do
+        HEADER="Comment from @$(jq --raw-output .comment.user.login "$GITHUB_EVENT_PATH"): "
+        BODY=$(jq --raw-output .comment.body "$GITHUB_EVENT_PATH" | parse_markdown)
+
+        echo "Adding comment to work item ${ID}..."
+        RESULTS=$(vsts work item update --id "$ID" --discussion "<p>${HEADER}</p>${BODY}")
+    done
+    ;;
+esac
```

Comment:
Add a default case, i.e. `*)`, maybe? To just log that there's no handling for the received events.